### PR TITLE
Added option to turn off dropout scaling

### PR DESCRIPTION
**Pull request by jfsantos**
In some implementations, one may wish to use dropout without feature scaling (i.e., dropout only masks features). I added a flag to the Dropout class and function to allow disabling scaling. Please let me know if you are interested in having this in the official repositories and I'll update the docs and tests accordingly so you can accept this PR.

### CODE REVIEW

**Comment by jfsantos** [comment]
I was just rechecking the [Dropout paper](http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf) and actually if we want to reproduce it we should not deactivate scaling, but only use it during test time (check section 4). The way this is implemented currently, scaling is applied no matter if the network is training or testing.

**Comment by unnonouno** [comment]
Thank you for PR and the reference. I'll check the original paper. Please wait.

**Comment by beam2d** [comment]
Thank you for the PR! Current implementation scales the values on training and does not on testing. This behavior is common to modern neural network implementations (e.g. Caffe does, Torch.nn also does in v2 dropout implementation). This is superior to the method of the original paper in following points:

1. We can obtain the regularization power with no additional overhead on evaluation.
2. The scale of output is kept with and without dropout.

Especially the second point is important to design a deep network architecture. Correct scaling is critical to achieve effective backprop learning, so the point 2 makes it easy to add dropout functions to existing architectures without breaking the scale. So I want to keep the current behavior as is. OTOH, I also understand that it is convenient to provide a way to reproduce the experiments in the original paper, so I agree on adding an option for it.

**Comment by jfsantos** [comment]
@beam2d, yes, I discussed this point this week here at the Deep Learning Summer School and it seems both kinds of implementations exist in the wild. In my use-case, I was trying to reproduce results which were achieved with the "original" Dropout, which led me to reading the paper, but I did not check other implementations and did not read any other papers proposing this scaled version. I guess this is how things work in deep learning, though :)

**Comment by beam2d** [comment]
I close this PR here since it is too old. If you want to add it to the current Chainer, please reopen it or make a new PR, where you should make the "scale-on-testing" mode optional (as you did in the first commit). Thank you!
